### ISSUE TITLE
Save the running results (dataframe) after the workflow completed

### ISSUE THREAD

**Issue by Marlin-Na** [getzlab/wolF#8]
I see the data frame are already generated https://github.com/getzlab/wolF/blob/master/wolf/workflow.py#L115. Think about saving it to current directory or NFS.

**Comment by julianhess** [comment]
Yup, it's already there, we just need to save it. I think the best place for it would be the root directory of the workflow on the NFS.

We don't ever want to save anything to the CWD.

**Comment by Marlin-Na** [comment]
One issue here: wolf is using context manager for running the workflows, it is not possible to return a data frame or path to the pickled data frame. Since one want to run it in a jupyter notebook session, one possible solution is to "temporarily" save with a global object/function. E.g. something like

```
from wolf import LastOutput
with Workflow() as w:
    w.run("a")
    w.run("b")
res = LastOutput()
```

**Comment by julianhess** [comment]
>it is not possible to return a data frame or path to the pickled data frame

Doesn't have to be returned; we just care that it's saved as the `results` property of the context manager object.

```python
with workflow() as w:
    w.run( ... )
    ...
```

results will be available in `w.results`.  Do you feel this is a bad UX?

**Comment by Marlin-Na** [comment]
Thanks for pointing out, this makes sense to me. I had a mistaken impression that variable will not be defined outside of context manager.

**Comment by julianhess** [comment]
I learned this fairly recently as well!

![image](https://user-images.githubusercontent.com/328077/74471981-2823e980-4e6f-11ea-9543-fa7d67a0397c.png)
